From 6910641acc13a6e58fe66753742c7a7034742933 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:04 +0800
Subject: [PATCH] arm64/mpam: free mon when remove momgroups
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6910641acc13a6e58fe66753742c7a7034742933


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/resctrlfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/resctrlfs.c b/fs/resctrlfs.c
index 255b91623392..d75004405803 100644
--- a/fs/resctrlfs.c
+++ b/fs/resctrlfs.c
@@ -774,6 +774,10 @@ static void resctrl_group_rm_mon(struct resctrl_group *rdtgrp,
 	struct resctrl_group *prdtgrp = rdtgrp->mon.parent;
 	int cpu;
 
+#ifdef CONFIG_ARM64 /* [FIXME] arch specific code */
+	free_mon(rdtgrp->mon.mon);
+#endif
+
 	/* Give any tasks back to the parent group */
 	resctrl_move_group_tasks(rdtgrp, prdtgrp, tmpmask);
 
-- 
2.26.2

